### PR TITLE
docs: improve escape pipe in Markdown tables

### PR DESCRIPTION
**Pull request by forresst**
From a remark on Crowdin:
> The value for the Type entry in the plugin-content-blog page is strange.
> In English documents, the Type item value is normally displayed.
> ex) editUrl: string | EditUrlFunction
> However, in French documentation, the Type item value is shown as an unknown code.
> ex) editUrl:!!crwdBlockTags_249_sgaTkcolBdwrc!!
> ex) blogSidebarCount: !!crwdBlockTags_250_sgaTkcolBdwrc!!

I think that the string `<code> 'light' &#124; 'dark' </code>` is blocked by Crowdin because of the pipe escape

This is a test to see if the other way to escape a pipe in a markdown table could solve the problem.

## Motivation

First, see if the preview of this page works well in Docusaurus.

In a second time, if the first point works, see if it unblocks the problem in Crowdin

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Related PRs

None

### CODE REVIEW

**Comment by forresst** [comment]
The other way to escape a pipe in a markdown table works:

image

Is it possible to merge this PR to see if it can solve the problem in Crowdin ?

**Comment by Josh-Cena** [comment]
Can we also fix this in other places? In `plugin-content-blog` and `plugin-content-docs`

**Comment by forresst** [comment]
> Can we also fix this in other places? In plugin-content-blog and plugin-content-docs

@Josh-Cena That's what I was thinking of doing in a second step, but I'd like to check that it fixes the problem in Crowdin first

**Comment by Josh-Cena** [comment]
Yes, but even if it doesn't, it improves readability (I didn't like the HTML entities either but didn't even think `<code> \|` as a possibility), and things are currently broken anyways, so it will be a gain with no loss

**Comment by Josh-Cena** [comment]
Yes, but even if it doesn't, it improves readability (I didn't like the HTML entities either but didn't even think `<code> \|` as a possibility), and things are currently broken anyways, so it will be a gain with no loss

**Comment by forresst** [comment]
@Josh-Cena Ok I will modify all the docs

File: website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md [hunk 1]
```diff
@@ -269,7 +269,7 @@ Accepted fields:
 | `to` | `string` | **Required** | Client-side routing, used for navigating within the website. The baseUrl will be automatically prepended to this value. |
 | `href` | `string` | **Required** | A full-page navigation, used for navigating outside of the website. **Only one of `to` or `href` should be used.** |
 | `prependBaseUrlToHref` | `boolean` | `false` | Prepends the baseUrl to `href` values. |
-| `position` | <code>'left' &#124; 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
+| `position` | <code>'left' \| 'right'</code | `'left'` | The side of the navbar this item should appear on. |
```
Comment (lex111) [inline website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md:272]:
```suggestion
| `position` | <code>'left' \| 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
```

File: website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md [hunk 2]
```diff
@@ -326,7 +326,7 @@ Accepted fields:
 | --- | --- | --- | --- |
 | `label` | `string` | **Required** | The name to be shown for this item. |
 | `items` | <code>[LinkLikeItem](#navbar-dropdown)[]</code> | **Required** | The items to be contained in the dropdown. |
-| `position` | <code>'left' &#124; 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
+| `position` | <code>'left' \| 'right'</code | `'left'` | The side of the navbar this item should appear on. |
```
Comment (lex111) [inline website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md:329]:
```suggestion
| `position` | <code>'left' \| 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
```

File: website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md [hunk 3]
```diff
@@ -374,7 +374,7 @@ Accepted fields:
 | --- | --- | --- | --- |
 | `docId` | `string` | **Required** | The ID of the doc that this item links to. |
 | `label` | `string` | `docId` | The name to be shown for this item. |
-| `position` | <code>'left' &#124; 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
+| `position` | <code>'left' \| 'right'</code | `'left'` | The side of the navbar this item should appear on. |
```
Comment (lex111) [inline website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md:377]:
```suggestion
| `position` | <code>'left' \| 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
```

File: website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md [hunk 4]
```diff
@@ -412,7 +412,7 @@ Accepted fields:
 
 | Name | Type | Default | Description |
 | --- | --- | --- | --- |
-| `position` | <code>'left' &#124; 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
+| `position` | <code>'left' \| 'right'</code | `'left'` | The side of the navbar this item should appear on. |
```
Comment (lex111) [inline website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md:415]:
```suggestion
| `position` | <code>'left' \| 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
```

File: website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md [hunk 5]
```diff
@@ -453,7 +453,7 @@ Accepted fields:
 | --- | --- | --- | --- |
 | `label` | `string` | The active/latest version label. | The name to be shown for this item. |
 | `to` | `string` | The active/latest version. | The internal link that this item points to. |
-| `position` | <code>'left' &#124; 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
+| `position` | <code>'left' \| 'right'</code | `'left'` | The side of the navbar this item should appear on. |
```
Comment (lex111) [inline website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md:456]:
```suggestion
| `position` | <code>'left' \| 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
```

File: website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md [hunk 6]
```diff
@@ -491,7 +491,7 @@ Accepted fields:
 
 | Name | Type | Default | Description |
 | --- | --- | --- | --- |
-| `position` | <code>'left' &#124; 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
+| `position` | <code>'left' \| 'right'</code | `'left'` | The side of the navbar this item should appear on. |
```
Comment (lex111) [inline website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md:494]:
```suggestion
| `position` | <code>'left' \| 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
```

File: website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md [hunk 7]
```diff
@@ -532,7 +532,7 @@ However, with this special navbar item type, you can change the default location
...
```
Comment (lex111) [inline website/versioned_docs/version-2.0.0-beta.6/api/themes/theme-configuration.md:535]:
```suggestion
| `position` | <code>'left' \| 'right'</code> | `'left'` | The side of the navbar this item should appear on. |
```

**Comment by lex111** [comment]
@forresst looks good to me, can I merge it?

**Comment by forresst** [comment]
@lex111 Perfect! Thanks! You can merge it!

**Review by lex111 — APPROVED** [review]

**Comment by lex111** [comment]
@forresst thanks!

**Comment by forresst** [comment]
Unfortunately, this still does not solve the problem in Crowdin 😢

**Comment by Josh-Cena** [comment]
My second guess is the `<code>` tag inside markdown table, is there a way to test Crowdin without updating the website? 🤔

**Comment by forresst** [comment]
@Josh-Cena I think you are right. the <code> tag can perhaps be problematic in a markdown table.

I don't know if it is possible to test it directly in Crowdin. An opinion @slorber ?

**Comment by lex111** [comment]
I'm not sure if Crowdin has a playground or something like that (FYI slorber on vacation).
Maybe we should change Type column to Possible Values, where we list comma-separated acceptable values?

| Name |  Possible Values | Default | Description |
| --- | --- | --- | --- |
| `defaultMode` | `'light'`, `'dark'` | `'light'` | The color mode when user first visits the site. |

**Comment by Josh-Cena** [comment]
@lex111 The original intent was to have the notation resemble TS—we have `Item[]`, `number | 'ALL'`, `string | EditUrlFunction`, etc., not sure if "possible values" would break the consistency

**Comment by lex111** [comment]
I misspelled, of course we are talking about types, not values. And it is unlikely that such change would worsen the understanding about Possible Types of fields. In our situation it would be a good compromise, so let's give it a try. We could even use just white space as a separator instead of comma.

**Comment by slorber** [comment]
Hey, created a dedicated issue and will ping Crowdin support about it: https://github.com/facebook/docusaurus/issues/5602

**Pull request merged**